stdlib/camlinternalComprehension.ml

```diff
@@ -1,11 +1,17 @@
 
-let map = List.map
+let map f y = 
```

Comment:
I think it is reasonable to change to the tail-recursive map. However, I think that means that there isn't any need to treat the single loop case specially, since it is going to be built and then reversed either way.